pyccl/core.py

```diff
@@ -182,6 +202,15 @@ def __init__(
 
         self._build_cosmo()
 
+        # TODO: include these in the self._params_init_kwargs?
+        # User input arrays:
```

Comment:
Might be a good place to check that if `background_on_input == True` then, all the input arrays are not `None` (and not `None`s).